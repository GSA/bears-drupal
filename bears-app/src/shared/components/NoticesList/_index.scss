#usagov-bears-app {
  .notices-list {
    margin-bottom: 32px;
  }

  .notice {
    margin-bottom: 24px;
  }

  .line-sperator-wrapper {
    padding-top: 32px;
    padding-bottom: 32px;
    margin: 32px;
  }

  .line-sperator {
    border-top: 2px solid lightgrey;
    width: 100%;
    height: 100%;
  }
}
