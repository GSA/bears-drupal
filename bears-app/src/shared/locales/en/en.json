{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
