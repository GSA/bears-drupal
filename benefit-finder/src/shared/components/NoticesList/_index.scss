@use '../../styles/colors/_index.scss' as color;
@use '../../styles/space/_index.scss' as space;
@use '../../styles/mixins/_index.scss' as *;

#benefit-finder {
  .notices-list {
    margin-bottom: space.$space-xl;
  }

  .notice {
    margin-bottom: space.$space-md-plus;
  }

  .notice-item {
    @include important-bold;
  }

  .line-sperator-wrapper {
    padding-top: space.$space-lg;
    padding-bottom: space.$padding-bottom-lg;
    margin: space.$space-lg;
  }

  .line-sperator {
    border-top: 2px solid color.$base-light-grey;
    width: 100%;
    height: 100%;
  }
}
