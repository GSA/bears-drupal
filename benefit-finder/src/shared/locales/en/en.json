{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
