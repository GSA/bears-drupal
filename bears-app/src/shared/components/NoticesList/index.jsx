import PropTypes from 'prop-types'
import createMarkup from '../../utils/createMarkup'
import { Alert } from '../index'

import './_index.scss'

/**
 * a functional component that renders a list of information
 * @component
 * @param {array} data inherited ui translations
 * @return {html} returns a semantic ul
 */
const NoticesList = ({ data }) => {
  /**
   * a functional component that renders a link as a button
   * @component
   * @param {object} data inherited ui translations
   * @return {html} returns a semantic li
   */
  const Notices = data => {
    return (
      data &&
      data.data.map((item, i) => {
        return (
          <li className="notice" key={`notice-${i}`}>
            <Alert noBackground>
              <div dangerouslySetInnerHTML={createMarkup(item.notice)}></div>
            </Alert>
          </li>
        )
      })
    )
  }

  return (
    <div className="notices">
      <ul className="notices-list add-list-reset">
        <Notices data={data} />
      </ul>
    </div>
  )
}

NoticesList.propTypes = {
  data: PropTypes.array,
}

export default NoticesList
