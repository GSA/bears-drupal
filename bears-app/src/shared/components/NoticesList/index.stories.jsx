import NoticesList from './index.jsx'

export default {
  component: NoticesList,
  tags: ['autodocs'],
}

export const Primary = {}
