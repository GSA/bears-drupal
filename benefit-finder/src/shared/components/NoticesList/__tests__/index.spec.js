import renderer from 'react-test-renderer'
import NoticesList from '../index.jsx'
import * as en from '../../../locales/en/en.json'

describe('NoticesList', () => {
  const t = en
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(
      <NoticesList data={t.intro.notices.list} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
