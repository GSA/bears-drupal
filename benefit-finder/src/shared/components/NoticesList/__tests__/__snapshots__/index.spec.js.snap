// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticesList renders a match to the previous snapshot 1`] = `
<div
  className="notices"
>
  <ul
    className="notices-list add-list-reset"
  >
    <li
      className="notice"
    >
      <div
        className=""
        role="alert"
      >
        <div
          className="usa-alert__body"
        >
          <div
            className="usa-alert__text"
          >
            <div
              className="notice-item"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<div>Your information<strong> is not</strong> shared, saved, or submitted.</div>",
                }
              }
            />
          </div>
        </div>
      </div>
    </li>
    <li
      className="notice"
    >
      <div
        className=""
        role="alert"
      >
        <div
          className="usa-alert__body"
        >
          <div
            className="usa-alert__text"
          >
            <div
              className="notice-item"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Provide complete info to get more relevant results.",
                }
              }
            />
          </div>
        </div>
      </div>
    </li>
    <li
      className="notice"
    >
      <div
        className=""
        role="alert"
      >
        <div
          className="usa-alert__body"
        >
          <div
            className="usa-alert__text"
          >
            <div
              className="notice-item"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>",
                }
              }
            />
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
